File: Beecrowd/PulaSapo/Sapo.java
/**
 * The Sapo class represents a frog that can jump over obstacles based on its jump height and the
 * height of the obstacle.
 */
package PulaSapo;

public class Sapo {
    private int jumpHeight;
    private int height = 1;

    Sapo(int jumpHeight) {
        this.jumpHeight = jumpHeight;
    }

    // *getters */

    public int getHeight() {
        return height;
    }

    public int getJumpHeight() {
        return jumpHeight;
    }

    // *setters */

    public void setHeight(int height) {
        this.height = height;
    }

    // *funções */

    /**
     * The function checks if an object can jump over an obstacle based on its height and jump height.
     * 
     * @param obstacle The parameter "obstacle" is of type "Obstaculo", which is a class representing
     * an obstacle.
     * @return The method is returning a boolean value.
     */
    public boolean jump(Obstaculo obstacle) {
        if ((this.height + this.jumpHeight) >= obstacle.getSize()
                && (Math.abs(this.height - obstacle.getSize())) <= this.jumpHeight) {
            this.height = obstacle.getSize();
            return true;
        } else
            return false;
    }
}

File: Beecrowd/PulaSapo/Sapo.py
# The Sapo class represents a character that can jump over obstacles based on its jump height and the
# size of the obstacle.


class Sapo:
    def __init__(self, jumpHeight):
        self.__jumpHeight = jumpHeight
        self.__height = 1

    # * getters

    def getJumpHeight(self) -> int:
        return self.__jumpHeight

    # * setters

    def setHeight(self, height) -> None:
        self.__height = height

    # * funções

    def jump(self, obstacle) -> bool:
        """
        The function checks if a character can jump over an obstacle based on their jump height and the
        obstacle's size.
        
        :param obstacle: The parameter "obstacle" represents an object of the class "Obstacle"
        :return: a boolean value. It returns True if the height of the obstacle is within the jump
        height range of the object, and False otherwise.
        """
        if (((self.__jumpHeight+self.__height) >= obstacle.getSize())
                and (abs(self.__height-obstacle.getSize()) <= self.__jumpHeight)):
            self.__height = obstacle.getSize()
            return True
        else:
            return False
